modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/ServerWorker.java

```diff
@@ -367,6 +368,7 @@ public void processNonEntityEnclosingRESTHandler(SOAPEnvelope soapEnvelope, Mess
                 AxisEngine.receive(msgContext);
             }
         } catch (AxisFault axisFault) {
+            consumeInputOnAxisFault(msgContext);
```

Comment:
let's move this consumeInputOnAxisFault call to handleException() method so that it will cover all the paths that this can happen.

Reply:
Fixed